server/api/crud/alerts.py

```diff
@@ -340,6 +341,14 @@ def _validate_alert(alert, name, project):
                 f"Invalid alert entity project ({alert.entities.project}) for alert {name} for project {project}"
             )
 
+    @staticmethod
+    def validate_alert_name(name):
+        pattern = re.compile(r"^[a-zA-Z0-9-]+$")
+        if not re.fullmatch(pattern, name):
```

Comment:
no need to compile the pattern separately if it’s only used once
```suggestion
        if not re.fullmatch(r"^[a-zA-Z0-9-]+$", name):
```